glyphs-reader/src/font.rs

```diff
@@ -1502,17 +1502,54 @@ fn default_master_idx(raw_font: &RawFont) -> usize {
                 .enumerate()
                 .find(|(_, master)| master.id == origin)
                 .map(|(idx, _)| idx)
-                .unwrap_or(0)
         })
-        // TODO: implement searching for "a base style shared between all masters" as glyphsLib does
-        // Still nothing? - just look for one called Regular
-        .or_else(|| {
-            raw_font
-                .font_master
-                .iter()
-                .position(|m| matches!(m.name.as_deref(), Some("Regular")))
+    {
+        return master_idx;
+    }
+    // Look for a master whose style is shared by all masters, defaulting to 'Regular'
+    let base_style = find_base_style(&raw_font.font_master).unwrap_or("Regular".to_string());
+    if let Some(master_idx) = raw_font
+        .font_master
+        .iter()
+        .position(|m| m.name.as_ref() == Some(&base_style))
+    {
+        return master_idx;
+    }
+    // Second try, this time omit 'Regular' from the master names
+    raw_font
+        .font_master
+        .iter()
+        .position(|m| {
+            m.name
+                .as_ref()
+                .map(|name| {
+                    name.split_whitespace()
+                        .filter(|n| *n != "Regular")
+                        .collect::<Vec<_>>()
+                        .join(" ")
+                })
+                .as_ref()
+                == Some(&base_style)
         })
-        .unwrap_or_default()
+        .unwrap_or_default() // just pick the first
+}
+
+// Direct port of glyphsLib's find_base_style:
```

Comment:
While true this doesn't actually help me understand what the relationship between master names and styles is and that makes interpreting the surrounding code difficult.

Reply:
I think we're saying the base style is the word(s) (defined as stuff delimited by whitespace) that are common in all master names is the base style, maybe we could comment as much and add an example.

Reply:
I did also find this pretty confusing, and the logic is also kind of funny but I guess depends highly on the use of standard naming conventions? Agree that a few examples would be useful...